### PR TITLE
feat(parsers): add support for RBS

### PR DESCRIPTION
**Pull request by apexatoll**
- This PR adds language support for the RBS typing language for Ruby.
- https://github.com/apexatoll/tree-sitter-rbs
- Changes
    - Add parser to parser list at `lua/nvim-treesitter/parsers.lua`
    - Add parser to README list
    - Copy queries into `queries/rbs`

- Please let me know if I have missed anything out!

### CODE REVIEW

**Comment by clason** [comment]
You need to add the parser to the `lockfile.json`.

File: queries/rbs/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+(var_name)    @variable
+(ivar_name)   @variable.instance
+(global_name) @variable.global
+
+(singleton_method      "self"  @variable.builtin)
+(singleton_method_name "self"  @variable.builtin)
+(module_function       "self?" @variable.builtin)
+
+[(symbol_literal) (attribute_name) (hash_key)] @symbol
```
Comment (ObserverOfTime) [inline queries/rbs/highlights.scm:36]: `(attribute_name)` should probably be `@attribute`.
`(hash_key)` should be `@label`.

File: queries/rbs/highlights.scm [hunk 2]
```diff
@@ -0,0 +1,42 @@
+["(" ")" "{" "}" "[" "]"] @punctuation.brackets
+["," "."]                 @punctuation.delimiter
+(method_signatures "|"    @punctuation.special)
+
+["class" "interface" "module" "end" "def" "alias"] @keyword.scope
```
Comment (ObserverOfTime) [inline queries/rbs/highlights.scm:5]:
```suggestion
["class" "interface" "module" "end" "def" "alias"] @keyword
```

File: queries/rbs/highlights.scm [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+(intersection_type "&")       @operator
+[":" "?"]                     @operator.special
+(splat_parameter "*")         @operator.special
+(double_splat_parameter "**") @operator.special
+
+["type" (visibility) (unchecked) (variance)] @keyword
+
+["include" "extend" "prepend"] @include
+
+[(class_name) (constant_name) (module_name) (interface_name)] @type.constant
```
Comment (ObserverOfTime) [inline queries/rbs/highlights.scm:19]: `(constant_name)` should be `@constant`.
`(module_name)` should be `@namespace`?
`(class_name)` and `(interface_name)` should be `@type`.

File: queries/rbs/highlights.scm [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+["=" "->" "<" "::"]           @operator
+(supertype)                   @operator
+(union_type "|")              @operator
+(intersection_type "&")       @operator
+[":" "?"]                     @operator.special
+(splat_parameter "*")         @operator.special
+(double_splat_parameter "**") @operator.special
+
+["type" (visibility) (unchecked) (variance)] @keyword
```
Comment (ObserverOfTime) [inline queries/rbs/highlights.scm:15]: Move "type" with the other keywords. The rest should probably be `@type.qualifier`.

File: queries/rbs/highlights.scm [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+["class" "interface" "module" "end" "def" "alias"] @keyword.scope
+
+["=" "->" "<" "::"]           @operator
+(supertype)                   @operator
+(union_type "|")              @operator
+(intersection_type "&")       @operator
+[":" "?"]                     @operator.special
+(splat_parameter "*")         @operator.special
+(double_splat_parameter "**") @operator.special
```
Comment (ObserverOfTime) [inline queries/rbs/highlights.scm:13]: ":" and "?" should be `@conditional.ternary` where applicable.
The rest should simply be `@operator`.

File: queries/rbs/highlights.scm [hunk 6]
```diff
@@ -0,0 +1,42 @@
+["(" ")" "{" "}" "[" "]"] @punctuation.brackets
+["," "."]                 @punctuation.delimiter
+(method_signatures "|"    @punctuation.special)
+
+["class" "interface" "module" "end" "def" "alias"] @keyword.scope
+
+["=" "->" "<" "::"]           @operator
+(supertype)                   @operator
+(union_type "|")              @operator
+(intersection_type "&")       @operator
```
Comment (ObserverOfTime) [inline queries/rbs/highlights.scm:10]:
```suggestion
(union_type "|"              @operator)
(intersection_type "&"       @operator)
```

**Review by ObserverOfTime — CHANGES_REQUESTED** [review]

File: queries/rbs/highlights.scm [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+(builtin_type)  @type.builtin
+(type_variable) @type.variable
+(alias_name)    @type.alias
+
+[(method_name) (singleton_method_name)] @function
+(attribute_type) @function.macro
+
+(var_name)    @variable
+(ivar_name)   @variable.instance
```
Comment (ObserverOfTime) [inline queries/rbs/highlights.scm:29]:
```suggestion
(ivar_name)   @property
```

File: queries/rbs/highlights.scm [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+["type" (visibility) (unchecked) (variance)] @keyword
+
+["include" "extend" "prepend"] @include
+
+[(class_name) (constant_name) (module_name) (interface_name)] @type.constant
+
+(builtin_type)  @type.builtin
+(type_variable) @type.variable
+(alias_name)    @type.alias
```
Comment (ObserverOfTime) [inline queries/rbs/highlights.scm:23]:
```suggestion
(type_variable) @type
(alias_name)    @type.definition
```

File: queries/rbs/indents.scm [hunk 9]
```diff
@@ -0,0 +1,4 @@
+[(class) (module) (interface)] @indent.begin
+
+"end" @indent.dedent
```
Comment (lucario387) [inline queries/rbs/indents.scm:3]: Don't see a point for this. Tldr is `@indent.dedent` is for all child nodes of the nodes defined, while `@indent.end` is for the final child node of an `@indent.end` block

File: queries/rbs/indents.scm [hunk 10]
```diff
@@ -0,0 +1,4 @@
+[(class) (module) (interface)] @indent.begin
```
Comment (lucario387) [inline queries/rbs/indents.scm:1]: Class, modules, interfaces, so there should be methods as well? And probably also branches, and such

**Pull request closed without merging**